pkg/backend/net/net.go

```diff
@@ -238,17 +243,25 @@ func (b Backend) do(trace types.Trace, req *http.Request) (string, []byte, error
 // If the backend timeout is positive, Call will override the context timeout
 // with the backend timeout.
 // Call ensures that the outgoing request has a UUID set.
-func (b Backend) call(ctx context.Context, trace types.Trace, u *url.URL) (string, []byte, error) {
+func (b Backend) call(ctx context.Context, trace types.Trace, u *url.URL, request string) (string, []byte, error) {
 	ctx, cancel := b.setTimeout(ctx)
 	defer cancel()
 
-	t0 := time.Now()
+	var t *prometheus.Timer
+	if b.qHist != nil { // TODO: remove condition when capi is merged with zipper
+		t = prometheus.NewTimer(b.qHist.WithLabelValues(request))
+		t.ObserveDuration()
+	}
+
 	err := b.enter(ctx)
-	trace.AddLimiter(t0)
 	if err != nil {
 		return "", nil, err
 	}
 
+	if t != nil {
```

Comment:
If I'm not mistaken, having this after the error check, we'll miss the times in queues which lead to error (e.g. timeout). I think we want to observe those times as well.

Reply:
I don't think we should. This is intentional but debatable.